draft-ietf-cose-hash-envelope.md

```diff
@@ -97,7 +93,7 @@ TBD_3:
 ~~~ cddl
 Hash_Envelope_Protected_Header = {
     ? &(alg: 1) => int,
-    ? &(payload_hash_alg: TBD_1) => int
+    &(payload_hash_alg: TBD_1) => int
```

Comment:
This is essential, as the hash algorithm MUST be signed to prevent hash algorithm substitution attacks.

The hash algorithm also conveys the length of the payload, for sha256, the payload will be 32 bytes.

The side of the preimage, is not relevant, assuming you trust the hash function.